haystack/preview/components/generators/openai/chatgpt.py

```diff
@@ -0,0 +1,267 @@
+            The callback function should accept two parameters: the token received from the stream and **kwargs.
+            The callback function should return the token to be sent to the stream. If the callback function is not
+            provided, the token is printed to stdout.
+        :param api_base_url: The OpenAI API Base url, defaults to `https://api.openai.com/v1`.
+        :param model_parameters: A dictionary of parameters to use for the model. See OpenAI
+            [documentation](https://platform.openai.com/docs/api-reference/chat) for more details. Some of the supported
+            parameters:
+            - `max_tokens`: The maximum number of tokens the output text can have.
+            - `temperature`: What sampling temperature to use. Higher values mean the model will take more risks.
+                Try 0.9 for more creative applications, and 0 (argmax sampling) for ones with a well-defined answer.
+            - `top_p`: An alternative to sampling with temperature, called nucleus sampling, where the model
+                considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens
+                comprising the top 10% probability mass are considered.
+            - `n`: How many completions to generate for each prompt. For example, if the LLM gets 3 prompts and n is 2,
+                it will generate two completions for each of the three prompts, ending up with 6 completions in total.
+            - `stop`: One or more sequences after which the LLM should stop generating tokens.
+            - `presence_penalty`: What penalty to apply if a token is already present at all. Bigger values mean
+                the model will be less likely to repeat the same token in the text.
+            - `frequency_penalty`: What penalty to apply if a token has already been generated in the text.
+                Bigger values mean the model will be less likely to repeat the same token in the text.
+            - `logit_bias`: Add a logit bias to specific tokens. The keys of the dictionary are tokens and the
+                values are the bias to add to that token.
+            - `openai_organization`: The OpenAI organization ID.
+
+        See OpenAI documentation](https://platform.openai.com/docs/api-reference/chat) for more details.
+        """
+        api_key = api_key if api_key is not None else self.api_key
+        if not api_key:
+            raise ValueError("OpenAI API key is missing. Please provide an API key.")
+
+        model_name = model_name or self.model_name
+        system_prompt = system_prompt if system_prompt is not None else self.system_prompt
+        model_parameters = model_parameters if model_parameters is not None else self.model_parameters
+        streaming_callback = streaming_callback or self.streaming_callback
+        api_base_url = api_base_url or self.api_base_url
```

Comment:
This are going to be remove but let's keep things consistent in the future. Using `or` in these cases is much simpler.

Reply:
In some cases falsy values are valid though... so I chose to use `or` where falsy values aren't possible, and a ternary operator where they are.